src/main/java/com/hubspot/jinjava/interpret/LazyExpression.java

```diff
@@ -0,0 +1,21 @@
+package com.hubspot.jinjava.interpret;
+
+import java.util.function.Supplier;
+
+public class LazyExpression implements Supplier {
+
+  private final Supplier supplier;
```

Comment:
Should this be `Supplier<?>`?

Reply:
Or perhaps `Supplier<Object>`. Maybe that's redundant.

Reply:
`Supplier` and `Supplier<Object>` will compile down to the same code due to how generics work in Java (IIRC) so it won't make a difference.

Reply:
This is also only used to return a generic object from the interpreter resolve method so I think this is fine.